chatkit-core/src/test/kotlin/com/pusher/chatkit/users/state/CurrentUserReceivedTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.pusher.chatkit.users.state
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.pusher.chatkit.state.ChatState
+import com.pusher.chatkit.state.CurrentUserReceived
+import com.pusher.chatkit.state.State
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class CurrentUserReceivedTest : Spek({
+
+    val simpleUser = UserInternalType(
+        id = "danielle",
+        createdAt = 1583422281000L,
+        updatedAt = 1583422302000L,
+        name = "daniellevass",
+        avatarUrl = null,
+        customData = null
+    )
+
+    describe("given initial state") {
+        val initialState = State.initial()
+
+        describe("when a current user is received") {
+            val currentUserReceived = CurrentUserReceived(simpleUser)
+            val newState = currentUserReceivedReducer(initialState, currentUserReceived)
+
+            it("then the state contains the current user") {
+                assertThat(newState.chatState.currentUser).isEqualTo(simpleUser)
+            }
+        }
+    }
+
+    describe("given a current user") {
+
+        val initialState = State(
+            ChatState(
+                currentUser = simpleUser,
+                joinedRoomsState = null
```

Comment:
In the other tests, we've been using e.g.

```
        val initialState = State.initial().with(
            ChatState.initial().with(
                currentUser = simpleUser
            )
        )
```

which means you don't have to worry about specifying other parts of the state - they will get initialised sensibly and then you just replace the bit you want to set for the test.

Reply:
Ah interesting - I had been setting them through the constructor in previous tests but this makes sense! Have updated.